exploration/selection-declaration.md

```diff
@@ -0,0 +1,296 @@
+   0    *    {{You have {$num1 :integer} ({$num2 :number minimumFractionDigits=1}) wildebeest.}}
+   one  0    {{ }}
+   one  one  {{ }}
+   one  *    {{ }}
+   // more cases for other locales that use two/few/many
+   *    0    {{ }}
+   *    one  {{ }}
+   *    *    {{ }}
+   ```
+
+4. As a user (especially as a translator), I don't want to have to modify
+   declarations and selectors to keep them in sync.
+   ```
+   .input {$num :number minimumFractionDigits=1}
+   .match {$num}
+   * {{Shouldn't have to modify the selector}}
+   ```
+   > Note that this is currently provided for by the spec.
+
+
+### Multiple Selection Conflicts
+If a user writes multiple selectors for the same operand, which one formats the placeholder?
+
+```
+.match {$num :integer} {$num :number minimumFractionDigits=2}
+* * {{Which selector formats {$num}?}}
+
+.match {$num :number minimumFractionDigits=2} {$num :integer}
+* * {{Which selector formats {$num}?}}
+```
+
+If both formats are needed in the message (presumably they are or why the selector), 
+how does one reference one or the other?
+
+### Selection Is Different From Format
+If a user wants to separate selection and formatting, 
+but selectors have a formatting side-effect,
+how do I write a selector that doesn't interfere with the formatting of the _same_ operand?
+(Note that users could do this using a `.local` declaration to split the operand)
+
+We don't support selection on dates in LDML45, but it's easy to conceptualize.
+How can a user write a _selector_ that doesn't mess up a _formatter_?
+
+```
+.input {$d :datetime skeleton=yMMMdjm}
+.match {$d :datetime month=numeric}
+1 {{Today is {$d} in cold cold {$d :datetime month=long}}}
+* {{Today is {$d}}}
+```
+
+## Requirements
+
+_What properties does the solution have to manifest to enable the use-cases above?_
+
+## Constraints
+
+_What prior decisions and existing conditions limit the possible design?_
+
+## Proposed Design
+
+_Describe the proposed solution. Consider syntax, formatting, errors, registry, tooling, interchange._
+
+## Alternatives Considered
+
+_What other solutions are available?_
+_How do they compare against the requirements?_
+_What other properties they have?_
+
+### Allow both local and input declarative selectors with immutability
+
+In this alternative, we modify the syntax to allow selectors to 
+annotate an input variable (as with `.input`)
+or assign a local variable (as with `.local`).
+Either annotation is immutable and results in a Duplicate Declaration error
+if it attempts to annotate a variable previously annotated.
+
+Example:
+```
+.match {$input :function} $local = {$input :function}
+* * {{This annotates {$input} and assigns {$local} a value.}}
+
+.match $local1 = {$input :function} $local2 = {$input :function2}
+* * {{This assigns two locals}}
+
+.input {$input :function}
+.local $local = {$input :function}
+.match {$input :function} {$local :function}
+* * {{This produces two duplicate declaration errors.}}
```

Comment:
Why? I only see one declaration for each of the two variables.

Reply:
`.input {$input}` and `.match {$input}` both "declare" `$input` in this design. Ditto the `.local` and later selector.